service/controller/resource/machinepoolconditions/conditionupgrading.go

```diff
@@ -25,42 +23,31 @@ func (r *Resource) ensureUpgradingCondition(ctx context.Context, machinePool *ca
 			conditions.UnexpectedConditionStatusErrorMessage(machinePool, aeconditions.UpgradingCondition))
 	}
 
-	// Set initial Upgrading condition status to false, since the MachinePool
-	// is just created.
-	if capiconditions.IsUnknown(machinePool, aeconditions.UpgradingCondition) {
-		err = conditions.MarkUpgradingNotStarted(machinePool)
-		if err != nil {
-			return microerror.Mask(err)
-		}
+	lastDeployedReleaseVersion, isLastDeployedReleaseVersionSet := machinePool.Annotations[annotation.LastDeployedReleaseVersion]
+	if !isLastDeployedReleaseVersionSet {
+		// Node pool is being created.
+		conditions.MarkUpgradingNotStarted(machinePool)
+		r.logConditionStatus(ctx, machinePool, aeconditions.UpgradingCondition)
+		r.logDebug(ctx, "ensured condition Upgrading, cluster is being created")
 		return nil
 	}
 
-	// Let's now check if desired versions are set and deployed.
+	// Let's now check if desired release version is deployed.
 	desiredReleaseVersion := key.ReleaseVersion(machinePool)
-	desiredAzureOperatorVersion := key.OperatorVersion(machinePool)
-
-	cluster, err := util.GetClusterFromMetadata(ctx, r.ctrlClient, machinePool.ObjectMeta)
-	if err != nil {
-		return microerror.Mask(err)
-	}
-
-	tenantClusterClient, err := r.tenantClientFactory.GetClient(ctx, cluster)
-	if err != nil {
-		return microerror.Mask(err)
-	}
+	desiredReleaseVersionIsDeployed := lastDeployedReleaseVersion == desiredReleaseVersion
```

Comment:
New instead of checking all the nodes and their versions, we simply check the last deployed version.